Add section about new Debug Page to Core Concepts > Cypress App (#4965)
cypress-io/cypress-documentation: docs/guides/core-concepts/cypress-app.mdx

for that change, and clicking on it will take you to the corresponding run page
in Cypress Cloud.

### Debug

<DocsImage
  src="/img/guides/core-concepts/cypress-app/debug-page.png"
  alt="Debug"
></DocsImage>

The Debug page allows you to easily debug failed CI test runs from within the
Cypress app. From this page, you can review, rerun, and debug failed CI test
runs that are recorded to Cypress Cloud (see [note above](#Overview)) – all
without leaving Cypress.

This feature eliminates the tedious process of switching between Cypress, your
local editor, and CI output, in order to fix failed tests. It does this by
enabling you to use the [Test Runner](#The-Test-Runner) to run only the tests
that failed in your last recorded test run, as well as review screenshots,
videos, and logs from your tests.

The Debug page shows the latest completed [test run](/guides/cloud/runs) that

> matches your current locally checked out git commit. If no run is found for your

Reviewer: I know we've gone around a bit on the terminology here. I'll throw out an alternative but no strong feelings either way:

```suggestion
matches the HEAD commit of your local project. If no run is found for your
```

> current commit, then Cypress crawls your git tree to find the most recent run in

Reviewer: While not inaccurate, `Cypress crawls your git tree` might sound a little more invasive than we'd like? Could just describe the end result here:

```suggestion
current commit, then Cypress shows the most recent run in
```

> your branch's history, including parent branches. The first 100 failed tests are

Reviewer: I think we could remove "including parent branches" here, as there's not really a parent/child relationship between branches in git. Once you label a commit with a branch, all history up until that point is considered part of that branch.

listed, with a link to Cypress Cloud to review the full run.

To use this feature, you must ensure that your version control is properly
configured and that you are sending git information to Cypress Cloud. Under some
environment setups (e.g. `docker`/`docker-compose`), you may have to pass git
information using

> [custom environment variables](/guides/continuous-integration/introduction#Git-information).

Reviewer: Do we want to add an explicit note that other version control systems aren't supported? Or do we assert that elsewhere already?

Reviewer 2: We typically only support git for all our features. I made it a bit more explicit by changing "version control" to "git" up above though.

Reviewer 2: We should probably make this more explicit throughout our docs though, this is a good callout. cc @ryanpei
